Lib/Search/ElasticSearch/Client.php

```diff
@@ -162,9 +163,32 @@ public function removeDocuments(ClassMetadata $class, array $documents)
      */
     public function removeAll(ClassMetadata $class, $query = null)
     {
-        $type = $this->getIndex($class->getIndexForRead())->getType($class->type);
+        $index = $class->getIndexForRead();
         $query = $query ?: new MatchAll();
-        $type->deleteByQuery($query);
+        $this->deleteByScanScroll($class, $query, $index);
+    }
+
+    /**
+     * Perform a scan and scroll through the results, bulk deleting by result ids
+     *
+     * @param ClassMetadata $class
+     * @param AbstractQuery $scanQuery
+     * @param string        $index
+     */
+    private function deleteByScanScroll(ClassMetadata $class, AbstractQuery $scanQuery, $index) {
+        $type = $this->getIndex($index)->getType($class->type);
+        $query = Query::create($scanQuery);
+        $results = $this->scan($query, [$class]);
+
+        foreach($results as $collection) {
+            $resultIds = array();
+            foreach($collection->getResults() as $result) {
+                $resultIds[] = $result->getId();
+            }
+            $this->client->deleteIds($resultIds, $index, $type);
+        }
+        // Refresh index to clear out deleted ID information
+        $this->getIndex($index)->refresh();
```

Comment:
removeAll() is only called in deleteByQuery() which is already deprecated. I meant you can just add the refresh option to EntityRepository::delete() similar to the save() method there. I think it's fine to change the method interface to be the same as save(), since I can't think of any reason to remove a document but not flush it. Also please update the doc block for it.